packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_login/ui_components/profile_avatar.dart

```diff
@@ -33,43 +39,78 @@ class ProfileAvatar extends StatelessWidget {
       );
 
   Widget _buildLoginButton(BuildContext context) => IconButton(
-        icon: Icon(context.designSystem.icons.login),
-        tooltip: context.l10n.featureLogin.logIn,
-        onPressed: () => context.router.push(const LoginRoute()),
-      );
+    icon: Icon(context.designSystem.icons.login),
+    tooltip: context.l10n.featureLogin.logIn,
+    onPressed: () =>
+        context.read<NavigationBlocType>().events.goTo(const LoginRoute()),
+  );
 
   Widget _buildLoggedInAvatar(BuildContext context) => PopupBuilder<String>(
-        tooltip: context.l10n.featureLogin.profile,
-        items: [
-          PopupMenuItem<String>(
-            value: _notificationKey,
-            child: Text(
-              context.l10n.featureLogin.notifications,
-              style: context.designSystem.typography.fadedButtonText,
-            ),
-          ),
-          const PopupMenuDivider(height: 2),
-          PopupMenuItem<String>(
-            value: _logoutKey,
-            child: Text(
-              context.l10n.featureLogin.logOut,
-              style: context.designSystem.typography.fadedButtonText,
-            ),
-          ),
-        ],
-        onSelected: (selected) {
-          if (selected == _notificationKey) {
-            context.router.push(const NotificationsRoute());
-          } else if (selected == _logoutKey) {
-            context.read<UserAccountBlocType>().events.logout();
-          }
-        },
-        child: Icon(
-          context.designSystem.icons.avatar,
-          color: Theme.of(context).colorScheme.onPrimary,
+    tooltip: context.l10n.featureLogin.profile,
+    items: [
+      PopupMenuItem<String>(
+        value: _notificationKey,
+        child: Text(
+          context.l10n.featureLogin.notifications,
+          style: context.designSystem.typography.fadedButtonText,
         ),
-      );
+      ),
+      const PopupMenuDivider(height: 2),
+      PopupMenuItem<String>(
+        value: _deepLinkFlow,
+        child: Text(
+          context.l10n.deepLinkFlowMenu,
+          style: context.designSystem.typography.fadedButtonText,
+        ),
+      ),
+      const PopupMenuDivider(height: 2),
+      PopupMenuItem<String>(
+        value: _pageWithResultKey,
+        child: Text(
+          context.l10n.pageWithResult,
+          style: context.designSystem.typography.fadedButtonText,
+        ),
+      ),
+      const PopupMenuDivider(height: 2),
+      PopupMenuItem<String>(
+        value: _logoutKey,
+        child: Text(
+          context.l10n.featureLogin.logOut,
+          style: context.designSystem.typography.fadedButtonText,
+        ),
+      ),
+    ],
+    onSelected: (selected) async {
+      if (selected == _notificationKey) {
```

Comment:
Please reduce the complexity of this code snipper and extract it as a private method

Reply:
Done